### PR TITLE
ref(profiling): add integration tests for profile_functions_metrics

### PR DESCRIPTION
**Pull request by viglia**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -384,6 +381,7 @@ module = [
     "sentry.snuba.metrics.datasource",
     "sentry.snuba.metrics.fields.base",
     "sentry.snuba.metrics.query_builder",
+    "sentry.snuba.profile_functions_metrics",
```
Comment (asottile-sentry) [inline pyproject.toml:384]: files should not get added to the ignore list

**Review by asottile-sentry — DISMISSED** [review]

File: src/sentry/snuba/profile_functions_metrics.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+    transform_alias_to_input_format=False,
+    sample=None,
+    has_metrics=False,
+    use_metrics_layer=False,
+    skip_tag_resolution=False,
+    extra_columns=None,
+    on_demand_metrics_enabled=False,
+    on_demand_metrics_type: MetricSpecType | None = None,
+    fallback_to_transactions=False,
+):
```
Comment (asottile-sentry) [inline src/sentry/snuba/profile_functions_metrics.py:46]: new code should be typed

File: src/sentry/snuba/profile_functions_metrics.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+            "isMetricsData": True,
+            "meta": result["meta"],
+        },
+        params["start"],
+        params["end"],
+        rollup,
+    )
+
+
+def top_events_timeseries(
```
Comment (shruthilayaj) [inline src/sentry/snuba/profile_functions_metrics.py:144]: just a heads up, if you plan to use top events for this using the events_stats endpoint, you'll need to add your dataset to this list https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/organization_events_stats.py#L247-L256

**Review by shruthilayaj — APPROVED** [review]
lgtm

**Pull request merged**